### PR TITLE
Split the Global Styles variations in two: default and premium

### PR DESCRIPTION
**Pull request by BogdanUngureanu**
Related to https://github.com/Automattic/dotcom-forge/issues/2261, https://github.com/Automattic/dotcom-forge/issues/2260

## Proposed Changes

* Split the default style variation from the Premium style variations in theme showcase and design picker

## Testing Instructions

* Use the live link from the comments
* Go to `site-setup/designSetup?siteSlug=your-site` (where the your-site is free)
* Click on a free theme and notice that style variations UI is split in two
* Resize your browser resolution to mobile and make sure that the UI looks good.
* Click on a Premium and Woo theme and notice that the UI is not split in two
* Repeat the same step with a site with a premium subscription and notice that all free, premium and Woo themes have the style variations combined in a single list. 
* Go to /themes (logged in and logged out)
* Make sure that for free sites, the UI is split in two for free themes
* Make sure that for free sites, the Ui is combined in a single list for Premium and Woo themes
* Make sure that for Premium sites the UI is combined for Free, Premium and Woo themes
* Make sure that the UI looks good on both mobile and desktop resolutions

### Theme showcase
**Free site, free theme**
Screenshot 2023-05-26 at 13 29 49
**Premium site, free theme**
Screenshot 2023-05-26 at 13 32 20

**Free site, free theme (mobile)**
Screenshot 2023-05-31 at 21 12 29

### Design Picker
**Free site, free theme**
Screenshot 2023-05-26 at 13 35 55

**Free site, free theme (mobile)**

Screenshot 2023-05-31 at 21 17 18

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Review by mmtr — COMMENTED** [review]
Nice job, this is quite a big improvement! It works well, but I think there are a few things we need to address first.

---

I think this breaks the "Premium" badge we display on the thumbnails when selecting a custom variation:

Before | After
--- | ---
Screenshot 2023-05-26 at 15 42 53 | Screenshot 2023-05-26 at 15 42 39

---

This text feels too wordy, is it really necessary? The original mockup didn't include it:

Mockup | This PR
--- | ---
Screenshot 2023-05-26 at 15 46 38 | Screenshot 2023-05-26 at 15 45 21

---

Seems the tiles are a bit misaligned with the text?

Mockup | This PR
--- | ---
Screenshot 2023-05-26 at 15 47 38 | Screenshot 2023-05-26 at 15 47 50

---

The font size in the sidebar feels smaller when compared to production and I think I like it more how it was before. Can we keep using the same font size?

Before | After
--- | ---
Screenshot 2023-05-26 at 15 53 16 | Screenshot 2023-05-26 at 15 53 22

---

Nitpick, non-blocking: On mobile, when the site is on a paid plan, the "Styles" header seems unnecessary. 

Before | After
--- | ---
Screenshot 2023-05-26 at 15 56 26 | Screenshot 2023-05-26 at 15 56 34

---

In the theme showcase, when the site is on a paid plan, there is a huge separation between the header and the styles grid. We can probably reduce it:

Before | After
--- | ---
Screenshot 2023-05-26 at 16 04 31 | Screenshot 2023-05-26 at 16 04 25

---

In the mobile view of the theme showcase, when the site is on a Free plan, I think the header takes a considerable amount of space. Can we maybe reuse the same layout that we use in the design picker, so all the styles are displayed in the same horizontal line?

Before | After
--- | ---
Screenshot 2023-05-26 at 16 08 19 | Screenshot 2023-05-26 at 16 07 31

**Comment by BogdanUngureanu** [comment]
> In the mobile view of the theme showcase, when the site is on a Free plan, I think the header takes a considerable amount of space. Can we maybe reuse the same layout that we use in the design picker, so all the styles are displayed in the same horizontal line?

I think it's better to keep the existing split mode because it's more clear and we describe better what style is free and what styles are paid. 

In the design picker I used that approach because there's enough space to use the same design, so had to use that approach.

**Comment by mmtr** [comment]
> In the design picker I used that approach because there's enough space to use the same design, so had to use that approach.

It doesn't look like there's enough space to me 🤔. By default (without scrolling), the preview only takes about the 25% of the screen. Previously, it was closer to the 50%. That's why I think we can probably be consistent with the design picker and display all styles in the same row (see my suggested mockup below).

Before (production ) | After (this PR) | Design picker (This PR) | My proposal
--- | --- | --- | ---
Screenshot 2023-05-30 at 13 56 23 | Screenshot 2023-05-30 at 13 50 36 | Screenshot 2023-05-30 at 13 44 46 | Screenshot 2023-05-30 at 13 54 32

**Comment by mmtr** [comment]
These two items remain unchanged, but didn't get your thoughts on them. Did you miss them?

- This text feels too wordy, is it really necessary? The original mockup didn't include it:

Mockup | This PR
--- | ---
Screenshot 2023-05-26 at 15 46 38 | Screenshot 2023-05-26 at 15 45 21


- Nitpick, non-blocking: On mobile, when the site is on a paid plan, the "Styles" header seems unnecessary. 

Before | After
--- | ---
Screenshot 2023-05-26 at 15 56 26 | Screenshot 2023-05-26 at 15 56 34

**Comment by BogdanUngureanu** [comment]
> These two items remain unchanged, but didn't get your thoughts on them. Did you miss them?

Strange, they should have been fixed. 🤔 

> It doesn't look like there's enough space to me 🤔. By default (without scrolling), the preview only takes about the 25% of the screen. Previously, it was closer to the 50%. That's why I think we can probably be consistent with the design picker and display all styles in the same row (see my suggested mockup below).

Fair enough. Maybe I can add the text at the bottom of the UI. :)

**Comment by BogdanUngureanu** [comment]
> These two items remain unchanged, but didn't get your thoughts on them. Did you miss them?

I've just checked again and I don't get that subheader text in the Theme showcase and Design picker. 🤔

File: packages/design-preview/src/components/style-variation.tsx [hunk 1]
```diff
@@ -9,35 +8,22 @@ interface StyleVariationPreviewsProps {
...
 	showOnlyHoverViewDefaultVariation,
 } ) => {
 	return (
 		<GlobalStylesVariations
 			globalStylesVariations={ variations as GlobalStylesObject[] }
 			selectedGlobalStylesVariation={ selectedVariation as GlobalStylesObject }
-			premiumBadge={
-				showGlobalStylesPremiumBadge && (
-					<PremiumBadge
-						className="design-picker__premium-badge"
```
Comment (mmtr) [inline packages/design-preview/src/components/style-variation.tsx:30]: This class is no longer used, so I think we can remove a few styles from the SCSS file

File: packages/global-styles/src/components/global-styles-variations/index.tsx [hunk 2]
```diff
@@ -113,28 +121,75 @@ const GlobalStylesVariations = ( {
...
+						{ ! splitPremiumVariations && premiumStyles }
+					</div>
+				</div>
+				{ splitPremiumVariations && (
+					<div className="global-styles-variations__type">
+						<div className="global-styles-variations__header">
+							<h2> { translate( 'Premium styles' ) } </h2>
+							<div>
+								<p>{ premiumStylesDescription }</p>
+							</div>
```
Comment (mmtr) [inline packages/global-styles/src/components/global-styles-variations/index.tsx]: Is the wrapping `div` really needed?
```suggestion
							<p>{ premiumStylesDescription }</p>
```

File: client/my-sites/theme/theme-style-variations/index.tsx [hunk 3]
```diff
@@ -1,41 +1,37 @@
...
-					{ translate( 'Styles' ) }
-					<PremiumBadge shouldHideTooltip />
-				</h2>
-				<p>{ description }</p>
-			</div>
+			{ !! description && <p>{ description }</p> }
+
 			<div className="theme__sheet-style-variations-previews">
 				<AsyncLoad
-					require="@automattic/design-preview/src/components/style-variation"
```
Comment (mmtr) [inline client/my-sites/theme/theme-style-variations/index.tsx:33]: I think this component is no longer used, can we remove it?

**Review by mmtr — APPROVED** [review]
Nice! It works well for me. I only left a few minor comments, but can be addressed in a follow-up.

I think we should also revisit the mobile layout, since there can a be a huge separation between the default style and the premium styles:

Screenshot 2023-05-31 at 18 09 16

File: packages/design-preview/src/components/style-variation.tsx (hunk 1 above)
Comment (BogdanUngureanu) [reply packages/design-preview/src/components/style-variation.tsx:30]: Oh that's right. I planned to remove it but I guess I forgot about it.

**Comment by BogdanUngureanu** [comment]
I've updated the description with some up to date screenshots.

Hey @lucasmendes-design can you also have a look at the proposed changes, especially on the mobile viewports? The changes were implemented based on the desktop design and based on the existing UI on mobile viewports.

**Comment by lucasmendes-design** [comment]
Hi folks, some notes about the experience here: 

- I think the global style thumbnail doesn't support any tags for now. It's strange placing a free or premium tag on it. This is in my backlog to create a thumbnail like the premium features in Canvas ([example](https://d.pr/i/tFsGev)).  
image

- In general, I see the opposite rationale here. Instead of highlighting with a badge the free option, I think it makes more sense to highlight the premium one. 
- I'm using the animation below which has only the star and also the full badge. 
https://github.com/Automattic/wp-calypso/assets/98343583/989f54bc-50b2-49a0-9add-7843f6c269df

- So, I think we could have something like this as a sticky header when users scroll down. 
image

- I'm not sure which one is the latest one but on mobile, the scroll is more natural than on desktop. So, I think it's fine having that. And mobile doesn't have hover, so we need to explain what this star is. 
image

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/8021142

Thank you @BogdanUngureanu for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/8021142).
